SU-HSDO/suhumsci
# [STN-59] Integrate Decanter 5.x into Sass Build

## Summary

Added Decanter as a dependency to our Humsci Basic frontend build. The 6.x version is in alpha, so I opted to go with the latest 5.x version. 

Based on feedback from Shea about the direction Decanter is going, I added the `node_modules` directory as an include path instead of including the Decanter `src/scss` directory. This means our imports will reference longer paths, but will make it more obvious where imports are importing from.

To test integration, I imported the `decanter-no-markup` file and used one of the variables from Decanter. I also imported the `brand-bar` component styles.

Additionally, this PR switches our build from using Node Sass to using Dart Sass which has the latest version of Sass which provides some powerful new features. Dart Sass compiles faster as well. The downside is that this new version of Sass is returning deprecation warnings for Sass code in Decanter which will generate a bunch on noise on each build. Something to consider.

## Steps to Test

1. Run `npm install` in the Humsci Basic directory 
1. Run `npm run build:sass` to compile the styles. You will see some deprecation warnings, but the styles should compile as expected. The cardinal red variable from Decanter should compile to the correct color value and the brand bar styles should show up in the compiled stylesheet.
1. Run `npm run watch` to test that the watch setup still works
1. From the root of the project, run `npm run build:css` and `npm run watch` to ensure that Grunt does not try to compile any styles in our Humsci Basic theme.

## See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
